libecho/src/AudioStream.h

```diff
@@ -0,0 +1,127 @@
+#ifndef ECHOCONNECT_AUDIOSTREAM_H
+#define ECHOCONNECT_AUDIOSTREAM_H
+
+#include <QtCore/QThread>
+#include <QtMultimedia/QAudioFormat>
+#include <QtMultimedia/QAudio>
+#include <QtMultimedia/QAudioInput>
+#include <QDebug>
+#include <QMetaMethod>
+
+constexpr int NOTIFY_INTERVAL = 32;
+constexpr double VOLUME = 1.0;
+
+struct AudioStreamInfo {
+    int bufferSize;
+    int notifyInterval;
+    int periodSize;
+    qreal volume;
+};
+
+
+class AudioStreamSignalsAndSlots : public QThread {
+protected slots:
+    virtual void handleStateChanged_slot(QAudio::State newState) = 0;
+
+    virtual void handleNotify_slot() = 0;
+};
+
+template<typename StreamType>
+class AudioStream : public AudioStreamSignalsAndSlots {
+public:
+    using StatusType = std::pair<int, qint64>;
```

Comment:
Maybe make it a real structure and document it.

Reply:
I'll certainly document it but haing it as distinct structure seems like an overkill.